File: 118-pascals-triangle/118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        
        vector<vector<int>> A;
        for(int line = 1;line<=numRows;line++)
        {
            int c= 1;
            vector<int> B;
            for(int i=1;i<=line;i++)
            {
                B.push_back(c);
                c = c*(line - i)/i;
            }
            A.push_back(B);
        }
        return A;
        
    }
};

File: 118-pascals-triangle/118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        
        List<List<Integer>> res = new ArrayList<List<Integer>>();
        List<Integer> cur, pre = null;
        for(int i=0;i<numRows;i++)
        {
             cur = new ArrayList<>();
            for(int j=0;j<=i;j++)
            {
                if(j ==0 || j==i)
                {
                    cur.add(1);
                }
                else
                {
                    cur.add(pre.get(j) + pre.get(j-1));
                }
                
            }
            pre = cur;
            res.add(cur);
        }
        
        return res;
    }
}